src/Data/PQueue/Prio/Max.hs

```diff
@@ -116,394 +112,5 @@ module Data.PQueue.Prio.Max (
-
--- | /O(n log n)/. Return all (key, value) pairs in descending order by key.
-toDescList :: Ord k => MaxPQueue k a -> [(k, a)]
-toDescList (MaxPQ q) = fmap (first' unDown) (Q.toAscList q)
-
--- | /O(n log n)/. Equivalent to 'toDescList'.
---
--- If the traversal order is irrelevant, consider using 'toListU'.
-toList :: Ord k => MaxPQueue k a -> [(k, a)]
-toList = toDescList
-
--- | /O(n)/. An unordered right fold over the elements of the queue, in no particular order.
-foldrU :: (a -> b -> b) -> b -> MaxPQueue k a -> b
-foldrU = foldrWithKeyU . const
-
--- | /O(n)/. An unordered right fold over the elements of the queue, in no particular order.
-foldrWithKeyU :: (k -> a -> b -> b) -> b -> MaxPQueue k a -> b
-foldrWithKeyU f z (MaxPQ q) = Q.foldrWithKeyU (f . unDown) z q
-
--- | /O(n)/. An unordered left fold over the elements of the queue, in no particular order.
-foldlU :: (b -> a -> b) -> b -> MaxPQueue k a -> b
-foldlU f = foldlWithKeyU (const . f)
-
--- | /O(n)/. An unordered left fold over the elements of the queue, in no particular order.
-foldlWithKeyU :: (b -> k -> a -> b) -> b -> MaxPQueue k a -> b
-foldlWithKeyU f z0 (MaxPQ q) = Q.foldlWithKeyU (\z -> f z . unDown) z0 q
-
--- | /O(n)/. An unordered traversal over a priority queue, in no particular order.
--- While there is no guarantee in which order the elements are traversed, the resulting
--- priority queue will be perfectly valid.
-traverseU :: (Applicative f) => (a -> f b) -> MaxPQueue k a -> f (MaxPQueue k b)
-traverseU = traverseWithKeyU . const
-
--- | /O(n)/. An unordered traversal over a priority queue, in no particular order.
--- While there is no guarantee in which order the elements are traversed, the resulting
--- priority queue will be perfectly valid.
-traverseWithKeyU :: (Applicative f) => (k -> a -> f b) -> MaxPQueue k a -> f (MaxPQueue k b)
-traverseWithKeyU f (MaxPQ q) = MaxPQ <$> Q.traverseWithKeyU (f . unDown) q
-
--- | /O(n)/. Return all keys of the queue in no particular order.
-keysU :: MaxPQueue k a -> [k]
-keysU = fmap fst . toListU
-
--- | /O(n)/. Return all elements of the queue in no particular order.
-elemsU :: MaxPQueue k a -> [a]
-elemsU = fmap snd . toListU
-
--- | /O(n)/. Equivalent to 'toListU'.
-assocsU :: MaxPQueue k a -> [(k, a)]
-assocsU = toListU
-
--- | /O(n)/. Returns all (key, value) pairs in the queue in no particular order.
-toListU :: MaxPQueue k a -> [(k, a)]
-toListU (MaxPQ q) = fmap (first' unDown) (Q.toListU q)
-
--- | /O(log n)/. @seqSpine q r@ forces the spine of @q@ and returns @r@.
---
--- Note: The spine of a 'MaxPQueue' is stored somewhat lazily. Most operations
--- take great care to prevent chains of thunks from accumulating along the
--- spine to the detriment of performance. However, 'mapKeysMonotonic' can leave
--- expensive thunks in the structure and repeated applications of that function
--- can create thunk chains.
-seqSpine :: MaxPQueue k a -> b -> b
-seqSpine (MaxPQ q) = Q.seqSpine q
+import Prelude ()
```

Comment:
What is this good for?

Reply:
I doubt it's good for much. It was here before #26, which made it *mostly* redundant. `mapKeysMonotonic` suspends O(n) work. Under certain circumstances, you might want to force that work to happen immediately or at a particular time.

Reply:
Oh, misread your comment. We don't want the `Prelude`, which exports various names that clash with ours.

Reply:
Oh, right, didn't think there.